packages/xo-server/src/xapi/index.js

```diff
@@ -2139,18 +2139,25 @@ export default class Xapi extends XapiBase {
   }
 
   @deferrable
-  async createBondedNetwork($defer, { bondMode, mac = '', pifIds, ...params }) {
+  async createBondedNetwork(
+    $defer,
+    { bondMode, mac = '', pifIds: masterPifIds, ...params }
+  ) {
     const network = await this.createNetwork(params)
     $defer.onFailure(() => this.deleteNetwork(network))
-    // TODO: test and confirm:
-    // Bond.create is called here with PIFs from one host but XAPI should then replicate the
-    // bond on each host in the same pool with the corresponding PIFs (ie same interface names?).
-    await this.call(
-      'Bond.create',
-      network.$ref,
-      map(pifIds, pifId => this.getObject(pifId).$ref),
-      mac,
-      bondMode
+
+    const pifsByHost = {}
+    masterPifIds.forEach(pifId => {
+      this.getObject(pifId).$network.$PIFs.forEach(pif => {
+        if (pifsByHost[pif.host] === undefined) {
+          pifsByHost[pif.host] = []
+        }
+        pifsByHost[pif.host].push(pif.$ref)
+      })
+    })
+
+    await asyncMap(pifsByHost, pifs =>
+      this.call('Bond.create', network.$ref, pifs, mac, bondMode)
```

Comment:
Same MAC for multiple bonds?

Reply:
Indeed this might be problematic, it slipped passed me during my review...

Reply:
In the current behavior, the `xo-web` client pass a list of host master's PIFs without passing a MAC address. So, the MAC address will be inherited from the primary_slave PIF(eg. `eth0`).  

But, the MAC property is exposed on `xo-server` API. 

We have two solutions in this case, the first solution is to not expose the mac address, the second is to change the mac property in the API to be able to pass a mac address by host.

Reply:
I'm not sure a user would want to specify a MAC address per host.
I personally  don't see the point.

Leaving the inheritance seems fine to me. What do you think?

Reply:
IMHO, make it as simple as possible and we'll add features when necessary.

Reply:
If it doesn't make sense to actually pass a MAC address here, we may want to explicitly pass `''` and add a comment explaining why. (So basically, Badr's first solution)